### ISSUE TITLE
Audio not supported

### ISSUE THREAD

**Issue by Jeremy1026** [KhaosT/homebridge-nest-cam#22]
Audio appears to no be supported with the video feeds. Is there a way, or plan, to add support?

**Comment by KhaosT** [comment]
Yeah audio is not supported. Not really sure I’m going to support it, as ffmpeg itself isn’t really flexible on handling this…

**Comment by leoneleone** [comment]
@KhaosT 
Thank you for the plugin, its very useful.
I'd like to put in my vote for audio support. With the new Nest Hello (doorbell), it seems imperative that audio support be added, otherwise the doorbell would have very limited compatibility in HomeKit.

**Comment by leoneleone** [comment]
@KhaosT 
Thanks for continuing to make this plugin better with each update. 

I really believe that 2 way audio support essential for the Nest Hello doorbell cam. 
Are there any methods we can test to get 2 way audio on the Nest Hello in particular?

I already use your plugin on a Nest Hello, in conjuction with homebridge-nest (motion sensor) and get ok results with it. Once Nest release specific api for the Nest Hello (Button/Doorbell press event), we can create a Doorbell Service with the relevant Characteristic.ProgrammableSwitchEvent.SINGLE_PRESS for the Nest Hello in homebridge-nest based on the future “Doorbell press event”. 

Audio Support is completely integral for the Nest Hello to function as a useful HomeKit Doorbell Camera.

Any thoughts on this?

**Comment by kitknox** [comment]
I did start looking into adding audio support.  The current NexusStream handling is generating a play request which is actually receiving the audio.  I dumped the raw OPUS packets and then converted them offline into PCM and was able to confirm playback.  In terms of getting this data into the FFmpeg pipeline the first problem is that FFmpeg doesn't support OPUS packets without them being inside another container.  I think the easiest thing would be to direct decode to PCM with libopus and then shove the PCM data back into FFMPEG for AAC conversion which the iOS ecosystem is going to be happier with.

Adding support for two-way audio would involve further Nest protocol support.  It looks like some of the hostnames for this are already coming back in JSON responses.  Not sure if that is more of the protobuf stuff like what is used for playback.

**Comment by ChrisKader** [comment]
@kitknox do the nest cams work with WWGA since WWN is no longer letting people create stuff? I would love to update this library to support the merged google accounts and along with audio but I am having trouble finding a "Supported Devices" section.

**Comment by rollinnerdy** [comment]
I would like to take a stab at this but I am struggling to find where the specification for the nest camera stream came from. Can you provide links the stream spec or where you got the proto file to generate the protocol sources.

**Comment by Brandawg93** [comment]
I've resolved the google authentication and am now looking into the audio. I would appreciate any help [here](https://www.npmjs.com/package/homebridge-nest-cam2).
